src/VisualStudio/CSharp/Impl/Options/Formatting/StyleViewModel.cs

```diff
@@ -781,6 +781,350 @@ class List<T>
     public T this[int i] { get { return _values[i]; } }
 }
 //]
+";
+
+        private readonly string s_arithmeticParenthesesIgnore = $@"
+class C
+{{
+    void M()
+    {{
+//[
+        // {ServicesVSResources.Keep_all_parentheses_in_colon}
+        var v = a + (b + c);
+//]
+    }}
+}}
+";
+
+        private readonly string s_arithmeticParenthesesRequireForPrecedenceClarity = $@"
+class C
+{{
+    void M()
+    {{
+//[
+        // {ServicesVSResources.Prefer_colon}
+        var v = a + (b * c);
+
+        // {ServicesVSResources.Over_colon}
+        var v = a + b * c;
+//]
+    }}
+}}
+";
+
+        private readonly string s_arithmeticParenthesesRemoveIfUnnecessary = $@"
+class C
+{{
+    void M()
+    {{
+//[
+        // {ServicesVSResources.Prefer_colon}
+        var v = a + b * c;
```

Comment:
these variale names are going to clash and therefore won't be highlighted properly. is there a way to solve that?

Reply:
I don't know what this means.

Reply:
the variable name is the same, that might cause tiny issues when presenting the code in the ide, like this:
![image](https://user-images.githubusercontent.com/11444821/37373544-547b46a8-2717-11e8-955f-416be2f4601d.png)

it would be great if it was possible to have these variables in different methods, but is it possible to have two of these //[ visible regions? would they be shown correctly?

Reply:
Pow(enh, grahms_number)